### DISCUSSION TITLE
State estimates drift with Flowdeck attached

### DISCUSSION THREAD

**Discussion by NageshEranki** [bitcraze/discussions#1033] [Q&A] [answered]
Hi

There are many groups in my class that use the Crazyflie drones with the flowdecks. Sometimes, the state estimates (for x/y positions) output in cfclient drift off to large values even when the drone is put stationary on a flat level surface. I repeated this test several times thinking it was a calibration error at startup but I always saw the same thing.

This drift in position estimates went away as soon as I repeated the test without a flowdeck attached. Clearly, the flowdeck is at fault here, so how can I go about fixing this? I'd go ahead and purchase a bunch but they're quite expensive.

Thanks

**Comment by knmcguire** [comment, marked as answer]
Hi! We have noticed some instabilities with the flowdeck as well, similar to what you are observing. A quick fix is to make sure that the flowdeck is mounted more closer to its body pcb and not to close to the ground. Also, you should try to have the flowdeck look at a good lightened surface and it shouldn't be too dark.

Giving that, the noticed flowdeck instabilities was a small reason why the October release was put on hold (we were also experiencing some deck detection problems along this one), so this month we are working on figure it out why the flowdeck is performing poorly. The problem is that there hasn't been any major changes in the last couple of releases on the flowdeck so we are a bit puzzled about why this would be the case. What you can try is to [flash older firmware releases ](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/#firmware-upgrade) (I would recommend trying 2022.12 ) and see if this makes a difference. 

The reason why the drift disappears when you remove the flowdeck, is that a different position estimator is automatically chosen (I think you can see that in the console output). The complementary filter is the default filter if no 'positioning deck' is attached. The complementary filter does not have an x and y position estimation ability so will put that on zero, but it will not be able to fly based on position commands. Only the extended kalman filter and unscented kalman filter are able to determine position. See this documentation for a bit more detail: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/sensor-to-control/state_estimators/.

In the mean time, we will be studing the reduced estimating quality of the flowdeck [in this ticket](https://github.com/bitcraze/crazyflie-firmware/issues/1329). It seems to be pure software base but we are looking at all aspects at the moment.
